### PR TITLE
Port deprecated Wizard component to new PF5 implementation

### PR DESCRIPTION
**Pull request by adamkankovsky**

### CODE REVIEW

**Comment by rvykydal** [comment]
@adamkankovsky is the PR ready for review or should I rather wait for tests passing?

**Comment by adamkankovsky** [comment]
> @adamkankovsky is the PR ready for review or should I rather wait for tests passing?

I dont have time to test is, so its draft. I look at this tomorow.

**Comment by rvykydal** [comment]
Seems like https://github.com/rhinstaller/anaconda-webui/pull/48 got merged into the commit accidentally somehow.

**Review by rvykydal — CHANGES_REQUESTED** [review]
Currently we don't allow moving forward (to a later step - we disable them) on the sidebar (for example from Welcome to Installation Destination). This does not hold with the PR. Seems that what `canJumpToStep` assured was not ported fully.
We should also create a test for this contidion.

**Comment by rvykydal** [comment]
> Currently we don't allow moving forward (to a later step - we disable them) on the sidebar (for example from Welcome to Installation Destination). This does not hold with the PR. Seems that what `canJumpToStep` assured was not ported fully. We should also create a test for this contidion - I'll do it.

The PR: https://github.com/rhinstaller/anaconda-webui/pull/66
Strangely the test passes for me locally on this PR. I can see the issue only when running your PR locally with `test/webui_testvm.py fedora-rawhide-boot --rsync`

**Comment by adamkankovsky** [comment]
> Currently we don't allow moving forward (to a later step - we disable them) on the sidebar (for example from Welcome to Installation Destination). This does not hold with the PR. Seems that what `canJumpToStep` assured was not ported fully. We should also create a test for this contidion - I'll do it.

It seems to me that the whole can jump thing was actually about not being able to jump to a step that is hidden, but in the current implementation it is handled a bit differently, so this situation shouldn't arise anyway

**Comment by rvykydal** [comment]
> > Currently we don't allow moving forward (to a later step - we disable them) on the sidebar (for example from Welcome to Installation Destination). This does not hold with the PR. Seems that what `canJumpToStep` assured was not ported fully. We should also create a test for this contidion - I'll do it.
> 
> The test PR: #66 Strangely the test passes for me locally on this PR. I can see the issue only when running your PR locally with `test/webui_testvm.py fedora-rawhide-boot --rsync` @adamkankovsky can you please rebase the PR on top of the merged test? I'd expect it to pass but let's see :)

Ok, so apparently I have problems building updates image for local tests run. Either obsolete image is built using some old .tar.xz or if I remove it I am getting errors with
```
error: Bad file: /home/rvykydal/work/git/rvykydal/anaconda-webui.1/anaconda-webui-1.20.g3305a966b.tar.xz: No such file or directory
```
. I guess I must clear something, I'll play with it more, just @KKoukiou FYI.

**Comment by KKoukiou** [comment]
> > > Currently we don't allow moving forward (to a later step - we disable them) on the sidebar (for example from Welcome to Installation Destination). This does not hold with the PR. Seems that what `canJumpToStep` assured was not ported fully. We should also create a test for this contidion - I'll do it.
> > 
> > 
> > The test PR: #66 Strangely the test passes for me locally on this PR. I can see the issue only when running your PR locally with `test/webui_testvm.py fedora-rawhide-boot --rsync` @adamkankovsky can you please rebase the PR on top of the merged test? I'd expect it to pass but let's see :)
> 
> Ok, so apparently I have problems building updates image for local tests run. Either obsolete image is built using some old .tar.xz or if I remove it I am getting errors with
> 
> ```
> error: Bad file: /home/rvykydal/work/git/rvykydal/anaconda-webui.1/anaconda-webui-1.20.g3305a966b.tar.xz: No such file or directory
> ```
> 
> . I guess I must clear something, I'll play with it more, just @KKoukiou FYI.

Hm, maybe try starting from a clean repo? `git clean -fdx` and re-create the updates.img?

**Review by KKoukiou — COMMENTED** [review]
It's obvious now from the new test (thansk @rvykydal) that this regressed the following:

Future steps should be disabled in the sidebar.

**Comment by rvykydal** [comment]
> git clean -fdx

@KKoukiou this works, thank you!

File: src/components/AnacondaWizard.jsx [hunk 1]
```diff
@@ -103,159 +109,168 @@ export const AnacondaWizard = ({ dispatch, storageData, localizationData, runtim
...
-            data: {
-                deviceData: storageData.devices,
-                diskSelection: storageData.diskSelection,
-                dispatch,
-                storageScenarioId,
-                setStorageScenarioId: (scenarioId) => {
-                    window.sessionStorage.setItem("storage-scenario-id", scenarioId);
-                    setStorageScenarioId(scenarioId);
-                }
+    const stepsOrder = useMemo(() => {
```
Comment (KKoukiou) [inline src/components/AnacondaWizard.jsx:112]: What happens if stepsOrder is not memoized? Did you try without?

Comment (adamkankovsky) [reply src/components/AnacondaWizard.jsx:112]: This is from the part where I tried to disable the next steps. I'm going to go through the code again to see if there's anything like that somewhere.

File: src/components/AnacondaWizard.jsx [hunk 2]
```diff
@@ -172,90 +178,97 @@ export const AnacondaWizard = ({ dispatch, storageData, localizationData, runtim
...
 
     const isStepFollowedBy = (earlierStepId, laterStepId) => {
         const earlierStepIdx = flattenedStepsIds.findIndex(s => s === earlierStepId);
         const laterStepIdx = flattenedStepsIds.findIndex(s => s === laterStepId);
-        return earlierStepIdx < laterStepIdx;
-    };
-
-    const canJumpToStep = (stepId, currentStepId) => {
-        return stepId === currentStepId || isStepFollowedBy(stepId, currentStepId);
+        if (laterStepIdx === -1) {
```
Comment (rvykydal) [inline src/components/AnacondaWizard.jsx:208]: If I understand it right: "any step is followed by a step which is not found" it seems a bit counterintuitive to me for the function name (and original purpose).

Comment (adamkankovsky) [reply src/components/AnacondaWizard.jsx]: I found a solution where we don't have to change the isStepFollowedBy function.

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you!

File: src/components/AnacondaWizard.jsx [hunk 3]
```diff
@@ -373,78 +383,68 @@ const Footer = ({
         }
     };
 
-    const goToPreviousStep = (activeStep, onBack, errorHandler) => {
+    const goToPreviousStep = () => {
         // first reset validation state to default
-        setIsFormValid(true);
+        setIsFormValid(false);
```
Comment (KKoukiou) [inline src/components/AnacondaWizard.jsx:388]: how is this change related?

Comment (KKoukiou) [reply src/components/AnacondaWizard.jsx:388]: anyhow that's correct change - but it looks like unrelated to this.

Comment (adamkankovsky) [reply src/components/AnacondaWizard.jsx]: Yes, its unrelated. I remove it, but i take a look at this and isStepFollowedBy after merge.
